opentelemetry-user-events-logs/src/logs/exporter.rs

```diff
@@ -1,95 +1,51 @@
 use eventheader::{FieldFormat, Level, Opcode};
-use eventheader_dynamic::EventBuilder;
+use eventheader_dynamic::{EventBuilder, EventSet, Provider};
 use opentelemetry::{otel_debug, otel_info};
-use std::borrow::Cow;
-use std::collections::HashMap;
-use std::fmt::Debug;
+use std::sync::Arc;
+use std::{fmt::Debug, sync::Mutex};
 
 use opentelemetry::{logs::AnyValue, logs::Severity, Key};
 use opentelemetry_sdk::error::{OTelSdkError, OTelSdkResult};
 use std::{cell::RefCell, str, time::SystemTime};
 
-/// Provider group associated with the user_events exporter
-pub type ProviderGroup = Option<Cow<'static, str>>;
-
 thread_local! { static EBW: RefCell<EventBuilder> = RefCell::new(EventBuilder::new());}
 
-/// Exporter config
-#[derive(Debug)]
-pub struct ExporterConfig {
-    /// keyword associated with user_events name
-    /// These should be mapped to logger_name as of now.
-    pub keywords_map: HashMap<String, u64>,
-    /// default keyword if map is not defined.
-    pub default_keyword: u64,
-}
-
-impl Default for ExporterConfig {
-    fn default() -> Self {
-        ExporterConfig {
-            keywords_map: HashMap::new(),
-            default_keyword: 1,
-        }
-    }
-}
-
-impl ExporterConfig {
-    pub(crate) fn get_log_keyword(&self, name: &str) -> Option<u64> {
-        self.keywords_map.get(name).copied()
-    }
-
-    pub(crate) fn get_log_keyword_or_default(&self, name: &str) -> u64 {
-        if self.keywords_map.is_empty() {
-            self.default_keyword
-        } else {
-            self.get_log_keyword(name).unwrap_or(self.default_keyword)
-        }
-    }
-}
-
 /// UserEventsExporter is a log exporter that exports logs in EventHeader format to user_events tracepoint.
 pub struct UserEventsExporter {
-    provider: eventheader_dynamic::Provider,
-    exporter_config: ExporterConfig,
+    provider: Mutex<Provider>,
+    event_sets: Vec<Arc<EventSet>>,
```

Comment:
This doesn't scale well - as it assumes that the tracepoint index maps to the level. I would have kept it configurable.

Reply:
we can change, when we decide to support custom tracepoints. There is no such capability today that is exposed publicly. When we do custom tracepoints, we'd need hash map or similar, but for now, vec! is fastest.

Do you see any perf/scale issue with this?

Reply:
> Do you see any perf/scale issue with this?

Not the perf issue, but scale. However, I don't have a strong opinion on this - it's fine if we address it later.

Reply:
I was referring to this exporter. It does not expose a public api to let user create more tracepoints than the 5 of them we create!